src/completions.jl

```diff
@@ -1,6 +1,38 @@
 import REPL.REPLCompletions, FuzzyCompletions
 
-const CompletionSuggetion = Dict{Symbol, String}
+const CompletionSuggetion = let
+  n2t = (
+    (:replacementPrefix, String),
+    # suggestion body
+    (:text, String),
+    (:type, String),
+    (:icon, String),
+    (:rightLabel, String),
+    (:leftLabel, String),
+    (:description, String),
+    (:descriptionMoreURL, String),
+    # for `getSuggestionDetailsOnSelect` API
+    (:detailtype, String)
+  )
+  NamedTuple{tuple(first.(n2t)...), Tuple{last.(n2t)...}}
+end
```

Comment:
At this point we should probably just make `CompletionSuggestion` an actual type.

Reply:
could you tell me why you think so ?

Reply:
okay, I didn't know JSON.jl can correctly `json` on `struct`s.
I removed those verbose `NamedTuple`s now.